When mix of fixable and in-fixable error,  `stdout` will throw error.

Hi I have upgrade to v14.0.2 and found that if I have a in-fixable rule e.g. `no-console` , a error will throw.

For example for this code

```ts
/* eslint no-console: "error" */
  console.log(2)
```

I would except to have to fix as
```ts
console.log(2)
```

<hr />

Here's the log/error I got
> 22:25:40[ERROR] Formatter 'eslint_d' error: console.log(2)

```
          22:34:55[DEBUG] Running formatters on /home/raydev/git/userscripts/src/dlsite-enable-select-text/config.ts: { "eslint_d" }
          22:34:55[INFO] Run eslint_d on /home/raydev/git/userscripts/src/dlsite-enable-select-text/config.ts
          22:34:55[DEBUG] Run command: { "eslint_d", "--fix-to-stdout", "--stdin", "--stdin-filename", "/home/raydev/git/userscripts/src/dlsite-enable-select-text/config.ts" }
          22:34:55[DEBUG] Run CWD: /home/raydev/git/userscripts
          22:34:55[INFO] eslint_d exited with code 1
          22:34:55[DEBUG] eslint_d stdout: { "console.log(2)", "" }
          22:34:55[DEBUG] eslint_d stderr: { "" }
          22:34:55[ERROR] Formatter 'eslint_d' error: console.log(2)

```

--- a/lib/forwarder.js
+++ b/lib/forwarder.js
@@ -65,28 +65,30 @@ export async function forwardToDaemon(resolver, config) {
       }
     })
     .on('end', () => {
-      if (fix_to_stdout) {
-        try {
-          const { output } = JSON.parse(flushMessage())[0];
-          process.stdout.write(output || text);
-        } catch (err) {
-          process.stdout.write(text);
-          console.error(`eslint_d: ${err}`);
-          process.exitCode = 1;
-          return;
-        }
-      }
-
       // The remaining 'content' must be the termination code:
       const match = content.match(EXIT_TOKEN_REGEXP);
-      if (match) {
+      if (!match) {
+        process.stdout.write(content);
+        console.error('eslint_d: unexpected response');
+        process.exitCode = 1;
+        return;
+      }
+
+      if (!fix_to_stdout) {
         process.exitCode = Number(match[1]);
         return;
       }
 
-      process.stdout.write(content);
-      console.error('eslint_d: unexpected response');
-      process.exitCode = 1;
+      try {
+        const { output } = JSON.parse(flushMessage())[0];
+        process.stdout.write(output || text);
+        // Exit code from eslint is deliberately ignored in this case
+        process.exitCode = 0;
+      } catch (err) {
+        process.stdout.write(text);
+        console.error(`eslint_d: ${err}`);
+        process.exitCode = 1;
+      }
     })
     .on('error', async (err) => {
       if (err['code'] === 'ECONNREFUSED') {

--- a/lib/forwarder.test.js
+++ b/lib/forwarder.test.js
@@ -277,7 +277,7 @@ describe('lib/forwarder', () => {
         socket.on.secondCall.callback(); // end
 
         assert.calledWith(process.stdout.write, 'response from daemon');
-        assert.equals(process.exitCode, 1);
+        assert.equals(process.exitCode, 0);
         refute.called(console.error);
       });
 
@@ -325,6 +325,26 @@ describe('lib/forwarder', () => {
           error = err;
         }
         assert.calledOnceWith(console.error, `eslint_d: ${error}`);
+      });
+
+      it('logs error and sets exitCode to 1 if response does not end with EXIT marker', async () => {
+        argv.push('--stdin', '--fix-to-stdout');
+        const chunks = ['response from daemon'];
+        sinon.replace(
+          socket,
+          'read',
+          sinon.fake(() => (chunks.length ? chunks.shift() : null))
+        );
+        sinon.replace(process.stdout, 'write', sinon.fake());
+
+        forwardToDaemon(resolver, config);
+        await new Promise(setImmediate);
+        socket.on.firstCall.callback(); // readable
+        socket.on.secondCall.callback(); // end
+
+        assert.calledWith(process.stdout.write, 'response from daemon');
+        assert.equals(process.exitCode, 1);
+        assert.calledOnceWith(console.error, 'eslint_d: unexpected response');
       });
     });
   });

--- a/test/test.integration.js
+++ b/test/test.integration.js
@@ -238,8 +238,7 @@ describe('integration tests', () => {
               `/* eslint radix: "error" */
             console.log('Hello' + parseInt('087'))`
             );
-            refute.isNull(error);
-            assert.equals(error?.['code'], 1);
+            assert.isNull(error);
           });
 
           it('prints fixed output if change is needed', async () => {
@@ -257,8 +256,7 @@ describe('integration tests', () => {
               `/* eslint radix: "error" */
             console.log('Hello' + parseInt('087'))`
             );
-            refute.isNull(error);
-            assert.equals(error?.['code'], 1);
+            assert.isNull(error);
           });
         });
       });

